### PR TITLE
(DOCSP-25095): Kotlin writeCopyTo()

### PR DESCRIPTION
**Pull request by cbullinger**
## Pull Request Info

Add **Copy Data into a New Realm** section to existing **Open a Realm - Kotlin SDK** page

### Jira

- https://jira.mongodb.org/browse/DOCSP-25095

### Staged Changes

- [Open a Realm](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-25095-copy-realm/sdk/kotlin/realm-database/realm-files/open-and-close-a-realm/#copy-data-into-a-new-realm)

### Reminder Checklist

If your PR modifies the docs, you might need to also update some corresponding
pages. Check if completed or N/A.

- [ ] Create Jira ticket for corresponding docs-app-services update(s), if any
- [x] Checked/updated Admin API
- [x] Checked/updated CLI reference

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/kotlin/realm-database/realm-files/open-and-close-a-realm.txt [hunk 1]
```diff
@@ -64,10 +62,68 @@ refer to :ref:`Open a Synced Realm <kotlin-open-a-synced-realm>`.
...
+You can copy data to a realm that uses the same configuration:  
+
+- Local realm to local realm
+- In-memory realm to in-memory realm
+- Synced realm to synced realm
+
+You can copy data to a realm that uses different sync configurations:
+
+- Local realm to Partition-Based Sync realm 
+- Synced realm to local realm
```
Comment (dacharyc) [inline source/sdk/kotlin/realm-database/realm-files/open-and-close-a-realm.txt:98]: I wonder if it's worth adding an example of copying a synced realm to a non-synced realm. In the Swift SDK, there is an additional consideration that isn't really covered here: [Open a Synced Realm as a Non-Synced Realm](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/docsp-25095-copy-realm/sdk/swift/sync/configure-and-open-a-synced-realm/#open-a-synced-realm-as-a-non-synced-realm)

Namely that copying a synced realm to a non-synced realm removes the `realm_id` in the non-synced realm, so you must increment the schema version as if you had deleted a property. Not sure if this is true in the Kotlin SDK, but may be worth exploring.

File: source/sdk/kotlin/realm-database/realm-files/open-and-close-a-realm.txt [hunk 2]
```diff
@@ -64,10 +62,68 @@ refer to :ref:`Open a Synced Realm <kotlin-open-a-synced-realm>`.
...
+unencrypted, in-memory synced realm to an encrypted local realm.
+
+Some additional considerations to keep in mind while using 
+``Realm.writeCopyTo()``: 
+
+- The destination file cannot already exist.
+- Copying a realm is not allowed within a write transaction or during
+  a :ref:`migration<kotlin-migrations>`.
+- When using Device Sync, you must sync all local changes with the
+  server before the copy is written. This ensures that the file can
```
Comment (dacharyc) [inline source/sdk/kotlin/realm-database/realm-files/open-and-close-a-realm.txt:114]: Does Kotlin have an API to wait for changes to upload? We might link to it here, if so. (I know Flutter does, but Swift does not expose this API.) Also, I wonder if it's worth adding an example of using writeCopy with a Synced realm to show the extra step of waiting for changes to upload?

**Review by dacharyc — APPROVED** [review]
Overall, LGTM. But a couple of Qs about adding details and/or possibly additional examples?

File: source/sdk/kotlin/realm-database/realm-files/open-and-close-a-realm.txt (hunk 1 above)
Comment (cbullinger) [reply source/sdk/kotlin/realm-database/realm-files/open-and-close-a-realm.txt:98]: I think you're right. I'll expand with a sync-related example. 

And the kotlin doc for [opening a synced realm](https://www.mongodb.com/docs/realm/sdk/kotlin/sync/open-a-synced-realm/) is pretty barebones, so it's probably worth it to expand the use cases/examples on that page in another ticket.

**Comment by edualonso** [comment]
Looks good 👍

**Pull request merged**